core/src/v2/schema.rs

```diff
@@ -54,6 +54,16 @@ pub trait Schema: Sized {
     /// - `serde_json::Value` works for both JSON and YAML.
     fn enum_variants(&self) -> Option<&[serde_json::Value]>;
 
+    /// More complex enum variants, where each has it's own schema.
+    fn any_of(&self) -> Option<&Vec<Resolvable<Self>>>;
```

Comment:
rather than just any_of I wonder if we could have a schema kind that would allow for all the other variants (oneOf, etc)

Reply:
example:
```rust
pub enum SchemaKind {
    Type(crate::v2::models::DataType),
    OneOf {
        #[serde(rename = "oneOf")]
        one_of: Vec<crate::v2::models::Resolvable<Self>>,
    },
    AllOf {
        #[serde(rename = "allOf")]
        all_of: Vec<crate::v2::models::Resolvable<Self>>,
    },
    AnyOf {
        #[serde(rename = "anyOf")]
        any_of: Vec<crate::v2::models::Resolvable<Self>>,
    },
}
```